src/pin.rs

```diff
@@ -16,37 +16,39 @@ use super::Register;
 ///
 /// The [`ExpanderInputPin`] instance can be used with other pieces of software using [`hal`].
 #[derive(Debug)]
-pub struct ExpanderInputPin<'a, I2C, Io>
+pub struct ExpanderInputPin<'a, I2C, Io, E>
```

Comment:
I don't think it is necessary to expose the E generic here and introduce a new phantomdata type. The E generic is already defined by the error type of the provided I2C generic.